lib/math/hash.flow

```diff
@@ -48,9 +57,33 @@ fastHashStringTailRecursion(key : string, len : int, cutoff : int, acc : int, in
 	} else {
 		fastHashStringTailRecursion(
 			key, len, cutoff, 
-			bitXor(acc, getCharCodeAt(key, ind)) * 1677761,
+			bitXor(acc, getCharCodeAt(key, ind)) * FNV_prime,
 			(ind + 1021) % len,
 			i + 1
 		)
 	}
 }
+
+addHashInt(hash: int, v: int) -> int {
+	hash1 = bitXor(hash, bitAnd(v, 0xFF)) * FNV_prime;
+	v1 = bitUshr(v, 8);
+	hash2 = bitXor(hash1, bitAnd(v1, 0xFF)) * FNV_prime;
+	v2 = bitUshr(v1, 8);
+	hash3 = bitXor(hash2, bitAnd(v2, 0xFF))* FNV_prime;
+	v3 = bitUshr(v2, 8);
+	bitXor(hash3, bitAnd(v3, 0xFF)) * FNV_prime;
+}
+
+addHashDouble(hash: int, d: double) -> int {
+	addHashBytes(hash, s2a(toBinary(flow(d))), 0);
+}
+
+addHashString(hash: int, s: string) -> int {
+	addHashBytes(hash, s2a(s), 0);
```

Comment:
Use getCharCodeAt and avoid s2a to do this hash without allocating in the heap.

Reply:
Done.